examples/hcha/hcha_trainer.py

```diff
@@ -0,0 +1,172 @@
+import os
+# os.environ['CUDA_VISIBLE_DEVICES']='1'
+os.environ['TL_BACKEND'] = 'torch'
+import sys
+# sys.path.insert(0, os.path.abspath('../../')) # adds path2gammagl to execute in command line.
+import argparse
+import tensorlayerx as tlx
+import numpy as np
+from gammagl.datasets import Planetoid
+# from gammagl.models import HCHA
+from gammagl.utils import add_self_loops, calc_gcn_norm, mask_to_index, set_device
+from tensorlayerx.model import TrainOneStep, WithLoss
+from gammagl.models import HCHA
+
+# tlx.set_device("GPU", 1)
+
+class SemiSpvzLoss(WithLoss):
+    def __init__(self, net, loss_fn):
+        super(SemiSpvzLoss, self).__init__(backbone=net, loss_fn=loss_fn)
+
+    def forward(self, data, y):
+        logits = self.backbone_network(data['x'], data['edge_index'], data['edge_weight'], data['edge_attr'])
+        train_logits = tlx.gather(logits, data['train_idx'])
+        train_y = tlx.gather(data['y'], data['train_idx'])
+        loss = self._loss_fn(train_logits, train_y)
+        return loss
+
+
+def calculate_acc(logits, y, metrics):
+    """
+    Args:
+        logits: node logits
+        y: node labels
+        metrics: tensorlayerx.metrics
+    Returns:
+        rst
+    """
+
+    metrics.update(logits, y)
+    rst = metrics.result()
+    metrics.reset()
+    return rst
+
+def main(args):
+    '''Data Preprocess'''
+    dataset = Planetoid(root='.', name=args.dataset)
+    dataset.process()
+    graph = dataset[0]
+    graph.tensor()
+    edge_index = graph.edge_index # Record which nodes have edges
+    edge_weight = tlx.ones((edge_index.shape[1],))  # Record each edge weights
+    x = graph.x  # Record features of each node
+    y = graph.y  # Record label of each node
+
+    '''label type'''
+    label_type = np.array(y)
+    label_type = set(list(label_type))
+
+    '''hyper-edge construction'''
+    temp = []
+    hedge_map = {}
+    for i in range(len(edge_index[0])):
+        if edge_index[0][i] not in temp:
+            temp.append(edge_index[0][i])
+            hedge_map[edge_index[0][i]] = [edge_index[0][i],edge_index[1][i]]
+        else:
+            hedge_map[edge_index[0][i]].append(edge_index[1][i])
+    hyperedge_index = [[],[]]
+    hyperedge_attr = []
+    for key, value in hedge_map.items():
+        m = np.zeros(len(x[0]))
+        count = 0
+        for item in value:
+            hyperedge_index[0].append(item) # node index
+            hyperedge_index[1].append(key) # hyperedge index
+            m += np.array(x[item])
+            count += 1
+        m = m/count
+        hyperedge_attr.append(m)
```

Comment:
Can we put these two ``for`` loop into one? And according to my testing, there is a bug here. I think ``tensor`` can not be treated as key in ``dict``.